src/anemoi/training/data/datamodule.py

```diff
@@ -74,11 +74,19 @@ def __init__(self, config: DictConfig) -> None:
             * self.config.hardware.num_nodes
             // self.config.hardware.num_gpus_per_model
         )  # number of model communication groups
+
```

Comment:
It would be nice to get model comm groups and model reader groups from the strategy / use the rountines in the strategy to compute them instead of having code here to re-compute the groups. This should be possible because we initialise the strategy before loading the datamodule.

Reply:
The datamodule only needed these to pass them down to the dataloader.dataset, I removed them entirely from the datamodule now.